src/assets/wise5/classroomMonitor/dataExport/strategies/AbstractComponentDataExportStrategy.ts

```diff
@@ -0,0 +1,393 @@
+import { AbstractDataExportStrategy } from './AbstractDataExportStrategy';
+
+export abstract class AbstractComponentDataExportStrategy extends AbstractDataExportStrategy {
+  canViewStudentNames: boolean;
+  includeOnlySubmits: boolean;
+  includeStudentNames: boolean;
+  workSelectionType: string;
+
+  constructor(protected nodeId: string, protected component: any, additionalParams: any) {
```

Comment:
Create new ComponentDataExportParams interface to type ```additionalParams```?
Type ComponentContent?
```suggestion
  constructor(protected nodeId: string, protected component: ComponentContent, additionalParams: ComponentDataExportParams) {
```